src/clouds/owncloud.js

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright (C) 2018 Orange.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Rombit - initial API and implementation
+ ******************************************************************************/
+
+// const debug = require('debug-levels')('agile-data');
+// const agile = require('./agile-sdk');
+// const get = require('lodash/get');
+const path = require('path');
+const request = require('request');
+
+module.exports = {
+  endpointDescription: function () {
+    return({
+      "cloud": "owncloud",
+    })
+  },
+  upload: function(credentials, remotepath,payload) {
+    return new Promise(function(resolve, reject) {
+
+      if (typeof remotepath !== 'undefined' && typeof remotepath !== "undefined"  && typeof payload !== "undefined") {
```

Comment:
It's better to check for `if (remotepath)` instead of `if (typeof remotepath !=='undefined')`, because the second check will return true if `remotepath` is `null`, and empty string, or 0.

Also `remotepath` is checked for twice.

Reply:
Fixed in the latest commit.